integrations/ollama/src/haystack_integrations/components/generators/ollama/chat/chat_generator.py

```diff
@@ -103,74 +106,107 @@ def _convert_ollama_meta_to_openai_format(intput_response_dict: Dict) -> Dict:
-                                })
-
-    messages = [ChatMessage.from_system("\nYou are a helpful, respectful and honest assistant"),
-    ChatMessage.from_user("What's Natural Language Processing?")]
-
-    print(generator.run(messages=messages))
+    from haystack.tools import Tool
+    def echo(query: str) -> str:
+        print("Tool executed with QUERY: {🔍 {query}}")
+        return f"🔍 {query}"
+    echo_tool = Tool(
+        name="echo_tool",
+        description="Echoes the query (demo tool).",
+        function=echo,
+        parameters={"query": {"type": "string", "description": "Search query"}},
+    )
+    agent = Agent(
+        chat_generator=OllamaChatGenerator(model="mistral-small3.1:24b"),
+        tools=[echo_tool],
+        system_prompt="Use tool to print the query to test tools. Do not answer the question, just send the query to the tool",
+        max_agent_steps=5,
+        raise_on_tool_invocation_failure=True,
+        streaming_callback=lambda ch: print(ch.content, end="", flush=True),
+    )
+    messages = [ChatMessage.from_user("This is stream test of tool usage")]
+    result = agent.run(messages=messages)
+    for message in result["messages"]:
+        print("\n======")
+        if message.role == "system":
+            continue
+        elif message.role == "tool":
+            print(f"{message.role}:")
+            print(f"Tool Results: {[tool.result for tool in message.tool_call_results]}")
+            print(f"Used Tools: {[tool.origin.tool_name for tool in message.tool_call_results]}\n")
+        else:
+            print(f"{message.role}: {message.text}")
+            print(f"Used Tools: {[tool.tool_name for tool in message.tool_calls]}\n")
     ```
     """
 
+    # ------------------------------------------------------------------
+    # Construction / (de)serialisation
+    # ------------------------------------------------------------------
     def __init__(
-        self,
-        model: str = "orca-mini",
-        url: str = "http://localhost:11434",
-        generation_kwargs: Optional[Dict[str, Any]] = None,
-        timeout: int = 120,
-        keep_alive: Optional[Union[float, str]] = None,
-        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
-        tools: Optional[Union[List[Tool], Toolset]] = None,
-        response_format: Optional[Union[None, Literal["json"], JsonSchemaValue]] = None,
+            self,
+            model: str = "orca-mini",
+            url: str = "http://localhost:11434",
+            generation_kwargs: Optional[Dict[str, Any]] = None,
+            timeout: int = 120,
+            keep_alive: Optional[Union[float, str]] = None,
+            streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+            tools: Optional[Union[List[Tool], Toolset]] = None,
+            response_format: Optional[Union[None, Literal["json"], JsonSchemaValue]] = None,
+            api_base_url: str | None = None,
```

Comment:
Thanks for looking to make things more consistent, but in favor of not making future breaking changes I'd say let's leave `url` and remove `api_base_url`

Reply:
done